beacon_node/beacon_chain/src/data_availability_checker.rs

```diff
@@ -543,6 +567,8 @@ pub struct AvailableBlock<E: EthSpec> {
     block_root: Hash256,
     block: Arc<SignedBeaconBlock<E>>,
     blobs: Option<BlobSidecarList<E>>,
+    /// Timestamp at which this block first became available (UNIX timestamp, time since 1970).
+    available_timestamp: Duration,
 }
```

Comment:
Storing the timestamp here will force you to include the timestamp for backfill & range-sync blocks which will only pollute the data. I think a better way would be to include this timestamp in the [`AvailableExecutedBlock`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/block_verification_types.rs#L230) & then populate it inside [`make_available()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker/overflow_lru_cache.rs#L117). This will probe what I believe you actually want as `make_available()` is the final destination of:
 - [`put_rpc_blobs()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker.rs#L203)
 - [`put_gossip_blob()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker.rs#L224)
 - [`put_pending_executed_block()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker.rs#L234)
 
 which are the 3 entry points for which new data can come in & complete a block.
 
 The other entry points:
 - [`verify_kzg_for_rpc_block()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker.rs#L247)
 - [`verify_kzg_for_rpc_blocks()`](https://github.com/sigp/lighthouse/blob/v5.1.3/beacon_node/beacon_chain/src/data_availability_checker.rs#L291)
 
 Don't actually hit the DA cache as they assume all blobs are already present.

Reply:
I tried doing this just now but got stuck because to fill in the `available_timestamp` on `AvailableExecutedBlock` here would require it to be part of the `AvailableBlock` anyway:

https://github.com/sigp/lighthouse/blob/3058b96f2560f1da04ada4f9d8ba8e5651794ff6/beacon_node/beacon_chain/src/block_verification_types.rs#L194-L200

Reply:
I tried doing this just now but got stuck because to fill in the `available_timestamp` on `AvailableExecutedBlock` here would require it to be part of the `AvailableBlock` anyway:

https://github.com/sigp/lighthouse/blob/3058b96f2560f1da04ada4f9d8ba8e5651794ff6/beacon_node/beacon_chain/src/block_verification_types.rs#L194-L200